gapic/schema/api.py

```diff
@@ -560,6 +561,116 @@ def mixin_http_options(self):
+        - The RPC must be a unary RPC (i.e. streaming RPCs are not supported)
+        - The field must not be annotated with google.api.field_behavior = REQUIRED.
+        - The field must be annotated with google.api.field_info.format = UUID4.
+        - The field presence requirements in AIP-4235 are checked at run time.
+
+        Args:
+            service_method_settings (Sequence[client_pb2.MethodSettings]): Method settings to be used
+                when generating API methods.
+        Return:
+            None
+        Raises:
+            ValueError: if fields in `method_settings.auto_populated_fields` cannot be
+                automatically populated.
+        """
+
+        all_errors: str = ""
+        invalid_selectors = []
+        for method_settings in service_method_settings:
+            method_descriptor = self.all_methods.get(method_settings.selector)
+            if not method_descriptor:
+                all_errors += f"Selector {method_settings.selector} is not a valid method in this API. "
+                invalid_selectors.append(method_settings.selector)
+                continue
+            if method_settings.auto_populated_fields:
+                # The method must not be a streaming method.
+                if method_descriptor.client_streaming or method_descriptor.server_streaming:
+                    all_errors += f"Selector {method_settings.selector} is a streaming method. `auto_populated_fields` are only supported for unary methods. "
+                    invalid_selectors.append(method_settings.selector)
+                    continue
+
+                selector_error_message: str = ""
+                top_level_request_message = self.messages[
+                    method_descriptor.input_type.lstrip(".")
+                ]
+                for field_str in method_settings.auto_populated_fields:
+                    if field_str not in top_level_request_message.fields:
+                        selector_error_message += f"Field `{field_str}` is not in the top level request message. "
+                    else:
+                        field = top_level_request_message.fields[field_str]
+                        if field.type != wrappers.PrimitiveType.build(str):
+                            selector_error_message += f"Field `{field_str}` is not of type string. "
+                        if field.required:
+                            selector_error_message += f"Field `{field_str}` is a required field. "
+                        if not field.uuid4:
+                            selector_error_message += f"Field `{field_str}` is not a UUID4 field. "
+
+                if selector_error_message:
+                    all_errors += selector_error_message
+                    invalid_selectors.append(method_settings.selector)
+
+        if all_errors:
+            raise ValueError(
+                f"Fields cannot be automatically populated in the top level "
```

Comment:
This error header text should be more general. For example, maybe the only error condition is the `not method_descriptor` above, and nothing in the config even intended to have auto-populated fields.

Reply:
Fixed in 5a2d33b8b160a246945e3752e145878581a351b1